src/App.js

```diff
@@ -1,79 +1,80 @@
 // import Heading from './components/Heading';
 import Dashboard from './pages/Dashboard';
+import useLocalStorage from './hooks/useLocalStorage';
 import { useState, useEffect } from 'react';
 
 export default function App() {
-  const [issues, setIssues] = useState([]);
-  const [loading, setLoading] = useState(null);
-  const [error, setError] = useState(null);
+  const [savedIssues, setSavedIssues] = useLocalStorage('savedIssues', []);
+  const [isLoading, setIsLoading] = useState(null);
+  const [hasError, setHasError] = useState(null);
 
   useEffect(() => {
-    GetFetch('https://api.github.com/repos/reactjs/reactjs.org/issues');
+    GetFetch('ttps://api.github.com/repos/reactjs/reactjs.org/issues');
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
   return (
     <Dashboard
-      issues={issues}
-      loading={loading}
-      error={error}
+      savedIssues={savedIssues}
+      isLoading={isLoading}
+      hasError={hasError}
       togglePin={togglePin}
+      GetFetch={GetFetch}
     />
   );
 
   function GetFetch(url) {
-    setLoading(true);
-    setIssues(null);
+    setIsLoading(true);
+    setHasError(false);
 
     const fetchData = async () => {
       try {
         const response = await fetch(url);
         if (response.ok) {
           const data = await response.json();
-          setIssues(
-            data.map(issue => {
-              const foundIssue = issues.find(
-                prevIssue => prevIssue.id === issue.id
-              );
-              if (foundIssue) {
-                return {
-                  ...issue,
-                  isPinned: foundIssue.isPinned,
-                };
-              } else {
-                return {
-                  ...issue,
-                  isPinned: false,
-                };
-              }
-            })
-          );
-          setLoading(false);
+          const fetchedData = data.map(issue => {
+            const foundIssue = savedIssues.find(
+              prevIssue => prevIssue.id === issue.id
+            );
+            if (foundIssue) {
+              return {
+                ...issue,
+                isPinned: foundIssue.isPinned,
+              };
+            } else {
+              return {
+                ...issue,
+                isPinned: false,
+              };
+            }
+          });
+          setSavedIssues(fetchedData);
+          setTimeout(() => setIsLoading(false), 2000);
         } else {
           throw new Error('Response not ok');
         }
       } catch (error) {
         console.error(error);
-        setLoading(false);
-        setError(true);
+        setIsLoading(false);
+        setHasError(true);
       }
     };
-    setTimeout(() => fetchData(), 1500);
+    fetchData();
```

Comment:
kannst du die function oben nicht gleich aufrufen statt diese erst in einer const zu speichern und dann aufzurufen? oder wird dazwischen noch was gemacht das ich hier übersehe?

Reply:
Oh yes, now it works